### PR TITLE
Minor adjustment: add managed policy to instance role

### PR DESCRIPTION
**Pull request by minchao**
Use managed policy `AmazonSSMManagedInstanceCore` instead of policy statement of instance role.

ref: https://docs.aws.amazon.com/systems-manager/latest/userguide/setup-instance-profile.html

### CODE REVIEW

File: src/spot.ts [hunk 1]
```diff
@@ -225,21 +225,11 @@ export class SpotFleet extends Resource {
...
-    // isntance role
-    this.instanceRole = props.instanceRole || new iam.Role(this, 'InstanceRole', {
-      roleName: PhysicalName.GENERATE_IF_NEEDED,
-      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
+    this.instanceRole = props.instanceRole || this.createInstanceRole();
+    this.instanceRole.addManagedPolicy({
+      managedPolicyArn: 'arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore',
     });
 
-    this.instanceRole.addToPolicy(new iam.PolicyStatement({
```
Comment (minchao) [inline src/spot.ts:234]: `addToPolicy`: deprecated method

File: src/spot.ts [hunk 2]
```diff
@@ -396,6 +386,13 @@ export class SpotFleet extends Resource {
     date.setSeconds(date.getSeconds() + duration.toSeconds());
     this.validUntil = date.toISOString();
   }
+
+  private createInstanceRole(): iam.Role {
```
Comment (minchao) [inline src/spot.ts:390]: hiding the implementation details to make it easy to read.

**Review by pahud — APPROVED** [review]
LGTM

**Comment by pahud** [comment]
ship it now! Thank you @minchao

**Pull request merged**